packages/dm-core-plugins/src/form/components/PrimitiveArray.tsx

```diff
@@ -39,7 +39,7 @@ const PrimitiveArray = ({
   const { getValues } = useFormContext()
 
   const updateValues = (index: number, newValue: TPrimitive): void => {
-    const newValues = [...getValues(namePath)]
+    const newValues = getValues(namePath) ? getValues(namePath) : []
```

Comment:
Pretty sure this should still be a copy, no? We're still modifying the array directly.
Also, shorthand `getValues(namepath) || []`? 🙂

Reply:
Will apply the shorthand 👍 
Modifying the return value of getValues does not cause a re-render, so it is not necessary to perform a deep copy.